app/controllers/concerns/current.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+module Current
+  thread_mattr_accessor :user
+
+  extend ActiveSupport::Concern
+  included { helper_method :set_current_user }
+
+  def set_current_user
+    Current.user = current_user
+    yield
+  ensure
+    Current.user = nil
+  end
+end
```

Comment:
looks strange, you can use just `current_user`
`Current` is absolutely unreadable, maybe make sense o rename it to `CurrentSession` and put here logic with current user and current order?

Reply:
I thought about it, but CurrentUser.user as for me was more weird.
I agree it looks strange, but my custom logic cannot use `current_user` in Form Object and Models (it's unavailable). That's way I decided to add this module, let it be `Current Session`.